pkg/module/parse/template_to_module.go

```diff
@@ -150,13 +154,22 @@ func (p *Parser) newManifestFromTemplate(
 		if err != nil {
 			return nil, err
 		}
-		componentDescriptor, err = p.ComponentDescriptorCache.GetRemoteDescriptor(ctx,
-			descriptorCacheKey, descriptor, p.Client)
+		componentDescriptor, err = parser.ComponentDescriptorCache.GetRemoteDescriptor(ctx,
+			descriptorCacheKey, descriptor, parser.Client)
 		if err != nil {
 			return nil, err
 		}
 	}
 
+	verification, err := signature.NewVerification(ctx,
+		parser.Client,
+		parser.EnableVerification,
+		parser.PublicKeyFilePath,
+		module.Name)
+	if err != nil {
+		return nil, err
+	}
+
 	if err := signature.Verify(componentDescriptor, verification); err != nil {
 		return nil, fmt.Errorf("could not verify descriptor: %w", err)
```

Comment:
I am not sure if the parser package will be also responsible to verify module signature. For me what the parser responsibility is to parse, unmarshal and deserlize the template. What do you think?

Reply:
Depending on the use case and the scope of the `parse`, this parse package defined here is responsible for parsing moduletemplate to manifest cr, and verification is the necessary middle step to be involved, if can't pass verification, then it wouldn't be converted to manifest cr, tbh, I don't see the responsible problem in this case.

Maybe the package name is a bit misleading, instead of `parse`, calling it `converter` is more suitable. But this can be a follow up issue if needed.

Reply:
Yep, if so its a bit misleading. Maybe a refactoring of the package name will be better.